config/cron-make-nightly-tarball.pl

```diff
@@ -87,9 +87,14 @@ sub doit {
 
     my $rc = system($cmd);
     if (0 != $rc && !$allowed_to_fail) {
-        # If we die/fail, ensure to change out of the temp tree so
-        # that it can be removed upon exit.
+        # If we die/fail, ensure to a) restore the git tree to a clean
+        # state, and b) change out of the temp tree so that it can be
+        # removed upon exit.
+        chdir($source_dir_arg);
+        system("git checkout .");
+        system("git clean -df");
```

Comment:
Good call; will do.

> On Feb 16, 2015, at 10:32 AM, Dave Goodell <notifications@github.com> wrote:
> 
> In config/cron-make-nightly-tarball.pl:
> 
> > @@ -87,9 +87,14 @@ sub doit {
> >  
> >      my $rc = system($cmd);
> >      if (0 != $rc && !$allowed_to_fail) {
> > -        # If we die/fail, ensure to change out of the temp tree so
> > -        # that it can be removed upon exit.
> > +        # If we die/fail, ensure to a) restore the git tree to a clean
> > +        # state, and b) change out of the temp tree so that it can be
> > +        # removed upon exit.
> > +        chdir($source_dir_arg);
> > +        system("git checkout .");
> > +        system("git clean -df");
> 
> I haven't read the whole script, but you probably want git clean -dxf (add -x) to make sure you really clean things up.
> 
> Same goes for the other git clean already in the script. That one could also benefit from a git reset --hard HEAD instead of git checkout . to make sure that you get the index back into a clean state too.
> 
> —
> Reply to this email directly or view it on GitHub.
> 


-- 
Jeff Squyres
jsquyres@cisco.com
For corporate legal information go to: http://www.cisco.com/web/about/doing_business/legal/cri/